### PR TITLE
Output a warning when running in a two-manager setup

### PR DESCRIPTION
**Pull request by thaJeztah**
Running a cluster in a two-manager configuration effectively *doubles* the chance of loosing control over the cluster (compared to running in a single-manager setup). Users may have the assumption that having two managers provides fault tolerance, so it's best to warn them if they're using this configuration.

This patch adds a warning to `docker info` if Swarm is configured with two managers:

    WARNING: Running Swarm in a two-manager configuration. This configuration provides
             no fault tolerance, and poses a high risk to loose control over the cluster.
             Refer to https://docs.docker.com/engine/swarm/admin_guide/ to configure the
             Swarm for fault-tolerance.

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

**- How to verify it**

run swarm in a two manager setup, and run `docker info`

**- Description for the changelog**

```
Output a warning when running in a two-manager setup
```

**- A picture of a cute animal (not mandatory but encouraged)**

panda-twins-cute-animals-pictures-pics

### CODE REVIEW

**Comment by dnephin** [comment]
I wonder if we should print this warning on `swarm join`? I guess it's fine on `info` as well, but it seems like it would be less visible there.

**Comment by thaJeztah** [comment]
Thought about `join` and `leave` / `demote`, but wondered if it would be misleading in cases where you'd go from `1 -> 2 -> 3` managers

**Comment by thaJeztah** [comment]
@dnephin opened an alternative that outputs a warning on;

- `docker info`
- `docker swarm join`
- `docker node demote`
- `docker node promote`

But not sure we need to do for all of those (also, the extra `/info` call for the `promote/demote` case may cause overhead, as all we need is the swarm info); https://github.com/docker/cli/compare/master...thaJeztah:warn-manager-count-alternative

**Comment by vdemeester** [comment]
ping @thaJeztah @dnephin @nishanttotla what is the status here ?

File: cli/command/system/info.go [hunk 1]
```diff
@@ -221,6 +221,15 @@ func prettyPrintInfo(dockerCli command.Cli, info types.Info) error {
 	fmt.Fprintln(dockerCli.Out(), "Live Restore Enabled:", info.LiveRestoreEnabled)
 	fmt.Fprint(dockerCli.Out(), "\n")
 
+	if getManagerCount(info.Swarm) == 2 {
+		msg := "WARNING: Running Swarm in a two-manager configuration. This configuration provides \n"
+		msg += "         no fault tolerance, and poses a high risk to loose control over the cluster.\n"
+		msg += "         Refer to https://docs.docker.com/engine/swarm/admin_guide/ to configure the\n"
+		msg += "         Swarm for fault-tolerance."
+
```
Comment (dnephin) [inline cli/command/system/info.go]: Nit, maybe use a raw string for this so it's easier to read?

**Comment by thaJeztah** [comment]
daemon-side implementation; https://github.com/moby/moby/pull/38891

**Comment by silvin-lubecki** [comment]
This PR needs to be rebased 🦁

**Comment by thaJeztah** [comment]
@silvin-lubecki I should probably close this one if we agree that https://github.com/moby/moby/pull/38891 is a better approach

**Comment by thaJeztah** [comment]
closing, as https://github.com/moby/moby/pull/38891 was merged

**Pull request closed without merging**